core/client/src/client.rs

```diff
@@ -691,6 +664,54 @@ impl<B, E, Block, RA> Client<B, E, Block, RA> where
 		Ok(ImportResult::Queued)
 	}
 
+	fn block_execution(
+		&self,
+		import_headers: &PrePostHeader<Block::Header>,
+		origin: BlockOrigin,
+		hash: Block::Hash,
+		body: Option<Vec<Block::Extrinsic>>,
+		transaction: &B::BlockImportOperation,
+	) -> error::Result<(Option<StorageUpdate<B, Block>>, Option<Option<ChangesUpdate>>, Option<Vec<(Vec<u8>, Option<Vec<u8>>)>>)> where
+		E: CallExecutor<Block, Blake2Hasher> + Send + Sync + Clone,
+	{
+		match transaction.state()? {
+			Some(transaction_state) => {
+				let mut overlay = Default::default();
+				let mut r = self.executor.call_at_state(
+					transaction_state,
+					&mut overlay,
+					"Core_execute_block",
+					&<Block as BlockT>::new(import_headers.pre().clone(), body.clone().unwrap_or_default()).encode(),
```

Comment:
I don't think `body` needs to be cloned here.

Reply:
Good catch, will fix

Reply:
resolved

Reply:
I think we should open an issue for that. I would like to see some function `Block::encode_from_parts(header: &Header, ext: &Vec<Extrensics>)` or something. This could be used here and would not require to `clone` the stuff.